charts/pmm/templates/statefulset.yaml

```diff
@@ -0,0 +1,122 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ include "pmm.fullname" . }}
+  labels:
+    {{- include "pmm.labels" . | nindent 4 }}
+spec:
+  serviceName: {{ .Values.service.name }}
+  replicas: 1
+  selector:
+    matchLabels:
+      {{- include "pmm.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "pmm.selectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "pmm.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      {{- with .Values.nodeSelector }}
+      nodeSelector:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.affinity }}
+      affinity:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.tolerations }}
+      tolerations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          envFrom:
+            - configMapRef:
+                name: {{ include "pmm.fullname" . }}
+                optional: true
+          env:
+            - name: GF_AUTH_GENERIC_OAUTH_CLIENT_ID
```

Comment:
@browseman I took those from infra examples, but I am not sure we need it and not sure if it works at all.

Reply:
As we already discussed this could be useful as an example for future customization that a user can make. Please be aware that the those two env. vars are not enough to integrate an internal iDP with Grafana. 

```
      - GF_AUTH_GENERIC_OAUTH_ENABLED='true'
      - GF_AUTH_GENERIC_OAUTH_SCOPES=''
      - GF_AUTH_GENERIC_OAUTH_AUTH_URL=''
      - GF_AUTH_GENERIC_OAUTH_TOKEN_URL=''
      - GF_AUTH_GENERIC_OAUTH_API_URL=''
      - GF_AUTH_GENERIC_OAUTH_ALLOWED_DOMAINS=''
      - GF_AUTH_GENERIC_OAUTH_CLIENT_ID=''
      - GF_AUTH_GENERIC_OAUTH_CLIENT_SECRET=''
```
The above is the min. required to get a generic OAuth working.

Reply:
right, I think it make sense to leave them (as they are secrets) and document that, and add other env as example under `pmmEnv`. Will do.